cocotb/log.py

```diff
@@ -136,10 +138,10 @@ def colour(self):
         )
         return want_color_output()
 
-    def setLevel(self, level: int) -> None:
+    def setLevel(self, level: typing.Union[int, str]) -> None:
         super().setLevel(level)
         if self.name == "gpi":
-            simulator.log_level(level)
+            simulator.log_level(self.level)
```

Comment:
Should this be the following? `Logger.level` is not documented.
```suggestion
            simulator.log_level(self.getEffectiveLevel())
```